zzk/test/zzkserver.go

```diff
@@ -0,0 +1,112 @@
+// Copyright 2015 The Serviced Authors.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build integration
+
+// Package zzktest provides helper code for integration tests that use Zookeeper
+package zzktest
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/control-center/serviced/dfs/docker"
+	dockerclient "github.com/fsouza/go-dockerclient"
+)
+
+// ZZKServer
+type ZZKServer struct {
+	Port     int
+	dc       *dockerclient.Client
+	zkCtrID  string
+}
+
+const (
+	DEFAULT_PORT = 2181
+	zzkVersion = "3.4.5"
+)
+
+// Start will start an instance of Zookeeper using a docker container.
+// If it finds that a container is running already, it will kill that container
+// before starting a new instance
+func (s *ZZKServer) Start() error {
+	var err error
+	if s.dc, err = dockerclient.NewClient(docker.DefaultSocket); err != nil {
+		return fmt.Errorf("Could not connect to docker client: %s", err)
+	}
+
+	// Make sure we start with a fresh instance
+	containerName :=  "zktestserver"
+	if ctr, err := s.dc.InspectContainer(containerName); err == nil {
+		fmt.Printf("ZZKServer.Start(): Killing container %s ...\n", ctr.ID)
+		s.dc.KillContainer(dockerclient.KillContainerOptions{ID: ctr.ID})
+		opts := dockerclient.RemoveContainerOptions{
+			ID:            ctr.ID,
+			RemoveVolumes: true,
+			Force:         true,
+		}
+		s.dc.RemoveContainer(opts)
+	} else {
```

Comment:
Is there a specific error we can check for here? Seems like we're catching them all to see if we have it.